plugin/technologymodel.cpp

```diff
@@ -316,6 +321,31 @@ void TechnologyModel::updateServiceList()
         Q_EMIT countChanged();
 }
 
+void TechnologyModel::removeService(const QString &path)
+{
+    int count;
+    int i;
+
+    if (m_techname.isEmpty())
+        return;
+
+    count = m_services.count();
+    for (i = 0; i < count ; i++) {
+        const NetworkService *s = m_services.value(i);
+
+        if (s->path() == path) {
+            disconnect(s, SIGNAL(destroyed(QObject*)),
+                   this, SLOT(networkServiceDestroyed(QObject*)));
+
+            beginRemoveRows(QModelIndex(), i, i);
+            m_services.remove(i);
+            endRemoveRows();
```

Comment:
I'm skeptical if or how much this would help the performance. If the updateServiceList() does something well, it's removing the old entries with one begin + endRemoveRows() as the last thing, while this does a separate operation pair for each removed service. It does help by making m_services shorter when the processing on the other method starts and avoids some looping but then introduces looping here instead.

Reply:
I dropped that change. I really have no idea how to improve this. All I think to "improve" this makes the process really complex and maybe not at all more effective...